app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDao.java

```diff
@@ -0,0 +1,147 @@
+        
+        // Get all the records because we only create a default if there are no physical records, 
+        // regardless of the deletion status.
+        List<DynamoSubpopulation> subpops = mapper.query(DynamoSubpopulation.class, query);
+        if (!allExistingRecordsOnly && subpops.isEmpty()) {
+            Subpopulation subpop = createDefaultSubpopulation(studyId);
+            return Lists.newArrayList(subpop);
+        }
+        // Now filter out deleted subpopulations, if requested
+        return subpops.stream()
+            .filter(subpop -> allExistingRecordsOnly || !subpop.isDeleted())
+            .sorted(SPECIFICITY_SORTER).collect(Collectors.toList());
+    }
+    
+    @Override
+    public Subpopulation createDefaultSubpopulation(StudyIdentifier studyId) {
+        DynamoSubpopulation subpop = new DynamoSubpopulation();
+        subpop.setStudyIdentifier(studyId.getIdentifier());
+        subpop.setGuid(studyId.getIdentifier());
+        subpop.setName("Default Consent Group");
+        subpop.setMinAppVersion(0);
+        subpop.setRequired(true);
+        mapper.save(subpop);
+        return subpop;
+    }
+    
+    @Override
+    public Subpopulation getSubpopulation(StudyIdentifier studyId, String guid) {
+        DynamoSubpopulation hashKey = new DynamoSubpopulation();
+        hashKey.setStudyIdentifier(studyId.getIdentifier());
+        hashKey.setGuid(guid);
+        
+        Subpopulation subpop = mapper.load(hashKey);
+        if (subpop == null) {
+            throw new EntityNotFoundException(Subpopulation.class);
+        }
+        return subpop;
+    }
+
+    @Override
+    public Subpopulation getSubpopulationForUser(ScheduleContext context) {
+        List<Subpopulation> found = Lists.newArrayList();
+
+        List<Subpopulation> subpops = getSubpopulations(context.getStudyIdentifier(), false);
+        for (Subpopulation subpop : subpops) {
+            boolean matches = CriteriaUtils.matchCriteria(context, subpop);
+            if (matches) {
+                found.add(subpop);
+            }
+        }
+        if (found.isEmpty()) {
+            throw new EntityNotFoundException(Subpopulation.class);
+        }
+        // Return the most specific match. Eventually we may return them all
+        return found.stream().sorted(SPECIFICITY_SORTER).collect(Collectors.toList()).get(0);
+    }
+    
+    @Override
+    public void deleteSubpopulation(StudyIdentifier studyId, String guid) {
+        Subpopulation subpop = getSubpopulation(studyId, guid);
+        subpop.setDeleted(true);
+        mapper.save(subpop);
+    }
+
+    @Override
+    public void deleteAllSubpopulations(StudyIdentifier studyId) {
+        List<Subpopulation> subpops = getSubpopulations(studyId, true);
+        
+        if (!subpops.isEmpty()) {
+            List<FailedBatch> failures = mapper.batchDelete(subpops);
+            BridgeUtils.ifFailuresThrowException(failures);
+        }
+    }
```

Comment:
As written, this has the unusual side effect that, if you call it on a study with no subpops, it will create a default subpop, then immediately delete it. That's one case for the getter not having side effects, or at the very least having a separate "createOnEmpty" flag.